### ISSUE TITLE
Error: Cannot find module './vendor/parse'

### ISSUE THREAD

**Issue by jacobmischka** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#405]
GPMDP works when running via npm, but the package doesn't work for me on Arch Linux x64 as of this morning. It worked last night.

```
[mischka@mischka-work Google-Play-Music-Desktop-Player-UNOFFICIAL-]$ npm run package:linux

> google-play-music-desktop-player@3.0.0 package:linux /home/mischka/projects/Google-Play-Music-Desktop-Player-UNOFFICIAL-
> gulp package:linux

[09:00:06] Using gulpfile ~/projects/Google-Play-Music-Desktop-Player-UNOFFICIAL-/gulpfile.js
[09:00:06] Starting 'clean-dist-linux'...
[09:00:06] Starting 'clean-external'...
[09:00:06] Starting 'clean-material'...
[09:00:06] Starting 'clean-utility'...
[09:00:06] Starting 'clean-internal'...
[09:00:06] Starting 'clean-images'...
[09:00:06] Starting 'clean-less'...
[09:00:06] Starting 'clean-fonts'...
[09:00:06] Starting 'clean-html'...
[09:00:06] Finished 'clean-external' after 11 ms
[09:00:06] Starting 'external'...
[09:00:06] Finished 'clean-material' after 61 ms
[09:00:06] Starting 'materialize-js'...
[09:00:06] Finished 'clean-utility' after 62 ms
[09:00:06] Starting 'utility-js'...
[09:00:06] Finished 'clean-html' after 65 ms
[09:00:06] Starting 'html'...
[09:00:06] Finished 'clean-less' after 67 ms
[09:00:06] Starting 'less'...
[09:00:06] Finished 'less' after 2.08 ms
[09:00:06] Finished 'clean-images' after 85 ms
[09:00:06] Starting 'images'...
[09:00:06] Finished 'clean-fonts' after 86 ms
[09:00:06] Starting 'fonts'...
[09:00:06] Finished 'materialize-js' after 221 ms
[09:00:06] Finished 'external' after 354 ms
[09:00:07] Finished 'utility-js' after 704 ms
[09:00:07] Finished 'html' after 698 ms
[09:00:07] Finished 'clean-dist-linux' after 1.12 s
[09:00:07] Finished 'images' after 1.03 s
[09:00:07] Finished 'clean-internal' after 1.12 s
[09:00:07] Starting 'transpile'...
[09:00:07] Finished 'transpile' after 2.83 ms
[09:00:08] Finished 'fonts' after 1.64 s
[09:00:08] Starting 'build'...
[09:00:08] Finished 'build' after 2.12 μs
[09:00:08] Starting 'package:linux'...
Packaging app for platform linux ia32 using electron v0.36.10
Packaging app for platform linux x64 using electron v0.36.10
[09:00:34] Finished 'package:linux' after 27 s
[mischka@mischka-work Google-Play-Music-Desktop-Player-UNOFFICIAL-]$ cd dist/Google\ Play\ Music\ Desktop\ Player-linux-x64/
[mischka@mischka-work Google Play Music Desktop Player-linux-x64]$ ./Google\ Play\ Music\ Desktop\ Player 
A JavaScript error occurred in the main process
Uncaught Exception:
Error: Cannot find module './vendor/parse'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:289:25)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Object.<anonymous> (/home/mischka/projects/Google-Play-Music-Desktop-Player-UNOFFICIAL-/dist/Google Play Music Desktop Player-linux-x64/resources/app/node_modules/parse-json/index.js:3:16)
    at Module._compile (module.js:425:26)
    at Object.Module._extensions..js (module.js:432:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Module.require (module.js:366:17)
```

For the record, since it looks suspicious, it worked after my commit.
```
[mischka@mischka-work Google Play Music Desktop Player-linux-x64]$ git checkout 949e95d787ff560a6bcad9ad7a268f8b2d7da631
[mischka@mischka-work Google-Play-Music-Desktop-Player-UNOFFICIAL-]$ npm run package:linux

> google-play-music-desktop-player@3.0.0 package:linux /home/mischka/projects/Google-Play-Music-Desktop-Player-UNOFFICIAL-
> gulp package:linux

[08:55:16] Using gulpfile ~/projects/Google-Play-Music-Desktop-Player-UNOFFICIAL-/gulpfile.js
[08:55:16] Starting 'clean-dist-linux'...
[08:55:16] Starting 'clean-external'...
[08:55:16] Starting 'clean-material'...
[08:55:16] Starting 'clean-utility'...
[08:55:16] Starting 'clean-internal'...
[08:55:16] Starting 'clean-images'...
[08:55:16] Starting 'clean-less'...
[08:55:16] Starting 'clean-fonts'...
[08:55:16] Starting 'clean-html'...
[08:55:16] Finished 'clean-external' after 12 ms
[08:55:16] Starting 'external'...
[08:55:16] Finished 'clean-material' after 60 ms
[08:55:16] Starting 'materialize-js'...
[08:55:16] Finished 'clean-utility' after 62 ms
[08:55:16] Starting 'utility-js'...
[08:55:16] Finished 'clean-html' after 67 ms
[08:55:16] Starting 'html'...
[08:55:16] Finished 'clean-less' after 69 ms
[08:55:16] Starting 'less'...
[08:55:16] Finished 'less' after 2.57 ms
[08:55:16] Finished 'clean-images' after 85 ms
[08:55:16] Starting 'images'...
[08:55:16] Finished 'clean-fonts' after 87 ms
[08:55:16] Starting 'fonts'...
[08:55:16] Finished 'materialize-js' after 242 ms
[08:55:16] Finished 'external' after 383 ms
[08:55:17] Finished 'utility-js' after 727 ms
[08:55:17] Finished 'html' after 719 ms
[08:55:17] Finished 'clean-dist-linux' after 1.18 s
[08:55:17] Finished 'images' after 1.09 s
[08:55:17] Finished 'clean-internal' after 1.18 s
[08:55:17] Starting 'transpile'...
[08:55:17] Finished 'transpile' after 2.64 ms
[08:55:18] Finished 'fonts' after 1.68 s
[08:55:18] Starting 'build'...
[08:55:18] Finished 'build' after 1.87 μs
[08:55:18] Starting 'package:linux'...
Packaging app for platform linux ia32 using electron v0.36.10
Packaging app for platform linux x64 using electron v0.36.10
[08:55:43] Finished 'package:linux' after 26 s
[mischka@mischka-work Google-Play-Music-Desktop-Player-UNOFFICIAL-]$ cd dist/Google\ Play\ Music\ Desktop\ Player-linux-x64/
[mischka@mischka-work Google Play Music Desktop Player-linux-x64]$ ./Google\ Play\ Music\ Desktop\ Player 
####################################
############### ERROR ##############
######### LASTFM  API KEYS #########
######## HAVE NOT BEEN SET #########
#####################################

```

**Comment by jacobmischka** [comment]
After bisecting, it broke on 6f989cd6fa39491b2d1f33ca4e4914f38af919de

**Comment by MarshallOfSound** [comment]
Have you run `npm install` recently?

**Comment by MarshallOfSound** [comment]
I added a new dependency in that commit

**Comment by jacobmischka** [comment]
Yes.

```
[mischka@mischka-work Google-Play-Music-Desktop-Player-UNOFFICIAL-]$ git bisect reset
Previous HEAD position was 5cc6344... Extract JSON file creation code to separate module
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
[mischka@mischka-work Google-Play-Music-Desktop-Player-UNOFFICIAL-]$ npm install[mischka@mischka-work Google-Play-Music-Desktop-Player-UNOFFICIAL-]$ npm run package:linux

> google-play-music-desktop-player@3.0.0 package:linux /home/mischka/projects/Google-Play-Music-Desktop-Player-UNOFFICIAL-
> gulp package:linux

[09:26:13] Using gulpfile ~/projects/Google-Play-Music-Desktop-Player-UNOFFICIAL-/gulpfile.js
[09:26:13] Starting 'clean-dist-linux'...
[09:26:13] Starting 'clean-external'...
[09:26:13] Starting 'clean-material'...
[09:26:13] Starting 'clean-utility'...
[09:26:13] Starting 'clean-internal'...
[09:26:13] Starting 'clean-images'...
[09:26:13] Starting 'clean-less'...
[09:26:13] Starting 'clean-fonts'...
[09:26:13] Starting 'clean-html'...
[09:26:13] Finished 'clean-external' after 13 ms
[09:26:13] Starting 'external'...
[09:26:13] Finished 'clean-material' after 61 ms
[09:26:13] Starting 'materialize-js'...
[09:26:13] Finished 'clean-utility' after 63 ms
[09:26:13] Starting 'utility-js'...
[09:26:13] Finished 'clean-less' after 66 ms
[09:26:13] Starting 'less'...
[09:26:13] Finished 'less' after 2.04 ms
[09:26:13] Finished 'clean-html' after 68 ms
[09:26:13] Starting 'html'...
[09:26:13] Finished 'clean-images' after 83 ms
[09:26:13] Starting 'images'...
[09:26:13] Finished 'clean-fonts' after 84 ms
[09:26:13] Starting 'fonts'...
[09:26:14] Finished 'materialize-js' after 203 ms
[09:26:14] Finished 'external' after 336 ms
[09:26:14] Finished 'utility-js' after 695 ms
[09:26:14] Finished 'html' after 687 ms
[09:26:14] Finished 'clean-dist-linux' after 1.11 s
[09:26:14] Finished 'images' after 1.02 s
[09:26:14] Finished 'clean-internal' after 1.11 s
[09:26:14] Starting 'transpile'...
[09:26:14] Finished 'transpile' after 2.82 ms
[09:26:15] Finished 'fonts' after 1.59 s
[09:26:15] Starting 'build'...
[09:26:15] Finished 'build' after 2.11 μs
[09:26:15] Starting 'package:linux'...
Packaging app for platform linux ia32 using electron v0.36.10
Packaging app for platform linux x64 using electron v0.36.10
[09:26:43] Finished 'package:linux' after 28 s
[mischka@mischka-work Google-Play-Music-Desktop-Player-UNOFFICIAL-]$ ls
appveyor.yml  docs               LICENSE            README.md
build         Gruntfile.js       MR_CHANGELOG.html  sig
circle.yml    gulpfile.js        node_modules       src
dist          ISSUE_TEMPLATE.md  package.json       vendor
[mischka@mischka-work Google-Play-Music-Desktop-Player-UNOFFICIAL-]$ dist/Google\ Play\ Music\ Desktop\ Player-linux-x64/Google\ Play\ Music\ Desktop\ Player 
A JavaScript error occurred in the main process
Uncaught Exception:
Error: Cannot find module './vendor/parse'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:289:25)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Object.<anonymous> (/home/mischka/projects/Google-Play-Music-Desktop-Player-UNOFFICIAL-/dist/Google Play Music Desktop Player-linux-x64/resources/app/node_modules/parse-json/index.js:3:16)
    at Module._compile (module.js:425:26)
    at Object.Module._extensions..js (module.js:432:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Module.require (module.js:366:17)
```

**Comment by MarshallOfSound** [comment]
Sometimes the nodes dependency tree screws up as well. You might be able to fix it by deleting the `node_modules` directory and re running `npm i`

**Comment by jacobmischka** [comment]
No dice. Removed node_modules and `npm install`ed. `npm start` works, `npm run package:linux && dist/Google\ Play\ Music\ Desktop\ Player-linux-x64/Google\ Play\ Music\ Desktop\ Player` doesn't.

**Comment by MarshallOfSound** [comment]
Hmmm, I'm going to have to look into this more.  But it's already late so I'm going to deal with it tomorrow :+1:

**Comment by jacobmischka** [comment]
The problem is with the package `parse-json`, a dependency of `yargs`. It uses a fallback if `JSON.parse()` isn't available, which is relatively required: `var fallback = require('.vendor/parse');`. I guess that doesn't work when using an electron package.

**Comment by MarshallOfSound** [comment]
@jacobmischka I just figured it out, I tried to package for windows and hit the same error, it's due to a bad regexp I set up recently in the `electron-packager` config

Fix incoming now

**Comment by MarshallOfSound** [comment]
Fixed in #214f399

**Comment by jacobmischka** [comment]
Nice, good work. :+1:
